tools/driver/driver.cpp

```diff
@@ -182,16 +207,27 @@ int main(int argc_, const char **argv_) {
   llvm::StringSaver Saver(Allocator);
   llvm::cl::ExpandResponseFiles(
       Saver,
-      llvm::Triple(llvm::sys::getProcessTriple()).isOSWindows() ?
-      llvm::cl::TokenizeWindowsCommandLine :
-      llvm::cl::TokenizeGNUCommandLine,
+      llvm::Triple(llvm::sys::getProcessTriple()).isOSWindows()
+          ? llvm::cl::TokenizeWindowsCommandLine
+          : llvm::cl::TokenizeGNUCommandLine,
       ExpandedArgs);
 
   // Initialize the stack trace using the parsed argument vector with expanded
   // response files.
   int ExpandedArgc = ExpandedArgs.size();
   const char **ExpandedArgv = ExpandedArgs.data();
+#ifdef _WIN32
+  // PROGRAM_START/InitLLVM overwrites the passed in arguments with UTF-8
+  // versions of them on Windows. This also has the effect of overwriting the
+  // response file expansion. Since we handle the UTF-8 conversion above, we
+  // pass in a copy and throw away the modifications.
+  SmallVector<const char *, 256> ExpandedArgs_ = ExpandedArgs;
+  int ExpandedArgc_ = ExpandedArgs_.size();
+  const char **ExpandedArgv_ = ExpandedArgs_.data();
+  PROGRAM_START(ExpandedArgc_, ExpandedArgv_);
+#else
   PROGRAM_START(ExpandedArgc, ExpandedArgv);
+#endif
```

Comment:
I think you can make this even simpler by renaming `ExpandedArgc/v` to `ThrowawayArgc/v`, and then initializing the `argv` ArrayRef below from `ExpandedArgs` instead. It's a good comment, though.

Reply:
Good call, +1 to 1 less ifdef _WIN32.